optimum/exporters/onnx/model_patcher.py

```diff
@@ -197,8 +284,42 @@ def patched_forward(*args, **kwargs):
             signature = inspect.signature(self.orig_forward)
             args, kwargs = override_arguments(args, kwargs, signature, model_kwargs=self.model_kwargs)
 
+            if _transformers_version >= version.parse("4.48"):
+                from transformers.cache_utils import DynamicCache, EncoderDecoderCache
+
+                if isinstance(kwargs.get("past_key_values"), (list, tuple)) and isinstance(
+                    kwargs["past_key_values"][0], (list, tuple)
+                ):
+                    if len(kwargs["past_key_values"][0]) == 2:
+                        kwargs["past_key_values"] = DynamicCache.from_legacy_cache(kwargs["past_key_values"])
+                    elif len(kwargs["past_key_values"][0]) == 4:
+                        kwargs["past_key_values"] = EncoderDecoderCache.from_legacy_cache(kwargs["past_key_values"])
+                    else:
+                        raise ValueError(
+                            f"past_key_values should have either 2 or 4 elements, but it has {len(kwargs['past_key_values'][0])} elements"
+                        )
+
+                elif any(isinstance(arg, (list, tuple)) for arg in args):
+                    for i in range(len(args)):
+                        if isinstance(args[i], (list, tuple)) and isinstance(args[i][0], (list, tuple)):
+                            if len(args[i][0]) == 2:
+                                args[i] = DynamicCache.from_legacy_cache(args[i])
+                            elif len(args[i][0]) == 4:
+                                args[i] = EncoderDecoderCache.from_legacy_cache(args[i])
+                            else:
+                                raise ValueError(
+                                    f"past_key_values should have either 2 or 4 elements, but it has {len(args[i])} elements"
+                                )
+                            break
+
             outputs = self.orig_forward(*args, **kwargs)
 
+            if _transformers_version >= version.parse("4.48"):
+                if "past_key_values" in outputs and isinstance(
+                    outputs["past_key_values"], (DynamicCache, EncoderDecoderCache)
+                ):
+                    outputs["past_key_values"] = outputs["past_key_values"].to_legacy_cache()
```

Comment:
Should be done on `filterd_outputs` directly instead of `outputs` (which is not always a dictionary) no ?